src/model/versioning.ts

```diff
@@ -207,7 +207,17 @@ export default class Versioning {
    * identifies the current commit.
    */
   static async getVersionDescription() {
-    return this.git(['describe', '--long', '--tags', '--always', 'HEAD']);
+    const versionTags = (await this.git(['tag', '--list', '--merged', 'HEAD', '--sort=-creatordate',]))
+      .split('\n')
+      .filter(tag => new RegExp(this.grepCompatibleInputVersionRegex).test(tag));
+    
+    if (versionTags.length === 0) {
+      core.warning('No valid version tags found. Using fallback description.');
+      return this.git(['describe', '--long', '--always', 'HEAD']);
```

Comment:
should this be `'--long', '--tags', '--always'`? Seems like tags is missing compared to the original version description.

Reply:
I'm asking because this should work in all scenarios on all platforms. So Linux, MacOS, Windows either with detached head and not. The current code is battle tested in that way.

Reply:
Thank you for these comments. You are right, all of them should be included.
Also, while checking it, I found that describe with the provided tag returns a commit count of 0, so I rewrote it a bit.